soccer/motion/MotionControl.cpp

```diff
@@ -59,25 +60,44 @@ void MotionControl::run() {
     ////////////////////////////////////////////////////////////////////
 
     float targetW = 0;
-    if (constraints.targetAngleVel) {
-        targetW = *constraints.targetAngleVel;
-    } else if (constraints.faceTarget || constraints.pivotTarget) {
-        Point targetPt = constraints.pivotTarget ? *constraints.pivotTarget
-                                                 : *constraints.faceTarget;
+    auto& rotationCommand = _robot->rotationCommand();
+    const auto& rotationConstraints = _robot->rotationConstraints();
+
+    boost::optional<Geometry2d::Point> targetPt;
+    const auto& motionCommand = _robot->motionCommand();
+    if (motionCommand->getCommandType() == MotionCommand::Pivot) {
+        PivotCommand command = *static_cast<PivotCommand*>(motionCommand.get());
+        targetPt = command.pivotTarget;
+    }
+    if (rotationCommand != nullptr) {
+        try {
+            Planning::FacePointCommand* facePointCommand =
+                dynamic_cast<Planning::FacePointCommand*>(
```

Comment:
`dynamic_cast` will return a `nullptr`, not throw a `bad_cast` upon failure in this case.  See http://en.cppreference.com/w/cpp/language/dynamic_cast